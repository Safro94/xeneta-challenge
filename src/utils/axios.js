import axios from 'axios';
import https from 'https';

const opts = {
	httpsAgent: new https.Agent({ rejectUnauthorized: false }),
	baseURL: process.env.REACT_APP_SERVER_URL,
};

const interceptorError = error => Promise.reject(error.response);

const instance = axios.create(opts);

instance.interceptors.response.use(response => response, interceptorError);
instance.defaults.headers.common['x-api-key'] = process.env.REACT_APP_API_KEY;

export default instance;
