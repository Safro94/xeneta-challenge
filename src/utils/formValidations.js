const validateSearchForm = ({
	departure,
	destination,
	departureDate,
	returnDate,
}) => {
	if (!departure || !destination || !departureDate) {
		return false;
	}

	if (departure.code === destination.code) {
		return false;
	}

	if (returnDate && returnDate < departureDate) {
		return false;
	}

	return true;
};

const errorCompare = (trip, message) => {
	const { departure, destination } = trip;

	const status = !departure && !destination ? true : false;

	return {
		message,
		status: !status && departure?.code === destination?.code,
	};
};

export { validateSearchForm, errorCompare };
