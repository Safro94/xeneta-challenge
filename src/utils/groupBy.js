import { marketPositions } from 'constants/marketPositions';

const groupBy = data =>
	data.reduce((accum, item) => {
		if (!accum[marketPositions.low]) accum[marketPositions.low] = [];
		if (!accum[marketPositions.average]) accum[marketPositions.average] = [];
		if (!accum[marketPositions.high]) accum[marketPositions.high] = [];

		accum[marketPositions.low].push({ x: item.day, y: item.low });
		accum[marketPositions.average].push({ x: item.day, y: item.mean });
		accum[marketPositions.high].push({ x: item.day, y: item.high });

		return accum;
	}, {});

export default groupBy;
