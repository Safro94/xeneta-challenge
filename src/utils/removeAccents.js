const removeAccents = value => {
	const map = {
		a: 'á|à|ã|â',
		e: 'é|è|ê',
		i: 'í|ì|î',
		o: 'ó|ò|ô|õ',
		u: 'ú|ù|û|ü',
	};

	for (let pattern in map) {
		value = value.replace(new RegExp(map[pattern], 'g'), pattern);
	}

	return value;
};

export { removeAccents };
