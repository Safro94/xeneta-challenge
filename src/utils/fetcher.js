import axios from 'utils/axios';

const fetcher = async ({ method = 'get', url, data = null }) => {
	return axios[method](url, JSON.parse(data)).then(
		response => response.data,
		error => {
			throw error;
		}
	);
};

export default fetcher;
