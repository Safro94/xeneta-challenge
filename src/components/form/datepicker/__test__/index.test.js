import { render } from 'utils/test-utils';
import Datepicker from '../';

describe('Datepicker', () => {
	const onSelectItem = jest.fn();
	const placeholderStart = 'Start date';
	const placeholderEnd = 'End date';

	it('should match the snapshot', () => {
		const { container } = render(
			<Datepicker
				onSelectItem={onSelectItem}
				placeholderStart={placeholderStart}
				placeholderEnd={placeholderEnd}
			/>
		);
		expect(container.firstElementChild).toMatchSnapshot();
	});
});
