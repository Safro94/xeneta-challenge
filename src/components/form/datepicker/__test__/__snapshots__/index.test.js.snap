// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Datepicker should match the snapshot 1`] = `
<div
  class="nice-dates"
>
  <div
    class="inputsContainer"
  >
    <input
      class="input"
      placeholder="Start date"
      type="text"
      value=""
    />
    <input
      class="input"
      placeholder="End date"
      type="text"
      value=""
    />
  </div>
  <div
    class="nice-dates-popover"
  >
    <div>
      <div
        class="nice-dates-navigation"
      >
        <a
          class="nice-dates-navigation_previous"
        />
        <span
          class="nice-dates-navigation_current"
        >
          febrero
        </span>
        <a
          class="nice-dates-navigation_next"
        />
      </div>
      <div
        class="nice-dates-week-header"
      >
        <span
          class="nice-dates-week-header_day"
        >
          lun
        </span>
        <span
          class="nice-dates-week-header_day"
        >
          mar
        </span>
        <span
          class="nice-dates-week-header_day"
        >
          mié
        </span>
        <span
          class="nice-dates-week-header_day"
        >
          jue
        </span>
        <span
          class="nice-dates-week-header_day"
        >
          vie
        </span>
        <span
          class="nice-dates-week-header_day"
        >
          sáb
        </span>
        <span
          class="nice-dates-week-header_day"
        >
          dom
        </span>
      </div>
      <div
        class="nice-dates-grid"
        style="height: 6px;"
      >
        <div
          class="nice-dates-grid_container -origin-top"
          style="transform: translate3d(0, 0px, 0); transition-duration: 500ms;"
        >
          <span
            class="nice-dates-day"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_month"
            >
              feb
            </span>
            <span
              class="nice-dates-day_date"
            >
              1
            </span>
          </span>
          <span
            class="nice-dates-day"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              2
            </span>
          </span>
          <span
            class="nice-dates-day"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              3
            </span>
          </span>
          <span
            class="nice-dates-day"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              4
            </span>
          </span>
          <span
            class="nice-dates-day"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              5
            </span>
          </span>
          <span
            class="nice-dates-day"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              6
            </span>
          </span>
          <span
            class="nice-dates-day -today"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              7
            </span>
          </span>
          <span
            class="nice-dates-day"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              8
            </span>
          </span>
          <span
            class="nice-dates-day"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              9
            </span>
          </span>
          <span
            class="nice-dates-day"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              10
            </span>
          </span>
          <span
            class="nice-dates-day"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              11
            </span>
          </span>
          <span
            class="nice-dates-day"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              12
            </span>
          </span>
          <span
            class="nice-dates-day"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              13
            </span>
          </span>
          <span
            class="nice-dates-day"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              14
            </span>
          </span>
          <span
            class="nice-dates-day"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              15
            </span>
          </span>
          <span
            class="nice-dates-day"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              16
            </span>
          </span>
          <span
            class="nice-dates-day"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              17
            </span>
          </span>
          <span
            class="nice-dates-day"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              18
            </span>
          </span>
          <span
            class="nice-dates-day"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              19
            </span>
          </span>
          <span
            class="nice-dates-day"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              20
            </span>
          </span>
          <span
            class="nice-dates-day"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              21
            </span>
          </span>
          <span
            class="nice-dates-day"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              22
            </span>
          </span>
          <span
            class="nice-dates-day"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              23
            </span>
          </span>
          <span
            class="nice-dates-day"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              24
            </span>
          </span>
          <span
            class="nice-dates-day"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              25
            </span>
          </span>
          <span
            class="nice-dates-day"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              26
            </span>
          </span>
          <span
            class="nice-dates-day"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              27
            </span>
          </span>
          <span
            class="nice-dates-day"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              28
            </span>
          </span>
          <span
            class="nice-dates-day -outside"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_month"
            >
              mar
            </span>
            <span
              class="nice-dates-day_date"
            >
              1
            </span>
          </span>
          <span
            class="nice-dates-day -outside"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              2
            </span>
          </span>
          <span
            class="nice-dates-day -outside"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              3
            </span>
          </span>
          <span
            class="nice-dates-day -outside"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              4
            </span>
          </span>
          <span
            class="nice-dates-day -outside"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              5
            </span>
          </span>
          <span
            class="nice-dates-day -outside"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              6
            </span>
          </span>
          <span
            class="nice-dates-day -outside"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              7
            </span>
          </span>
          <span
            class="nice-dates-day -outside"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              8
            </span>
          </span>
          <span
            class="nice-dates-day -outside"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              9
            </span>
          </span>
          <span
            class="nice-dates-day -outside"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              10
            </span>
          </span>
          <span
            class="nice-dates-day -outside"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              11
            </span>
          </span>
          <span
            class="nice-dates-day -outside"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              12
            </span>
          </span>
          <span
            class="nice-dates-day -outside"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              13
            </span>
          </span>
          <span
            class="nice-dates-day -outside"
            style="height: 1px;"
          >
            <span
              class="nice-dates-day_date"
            >
              14
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
